qp_qiime2/qiime2.py

```diff
@@ -24,6 +24,84 @@
 
 ALPHA_PHYLOGENETIC_METRICS = {'faith_pd'}
 
+ALPHA_DIVERSITY_METRICS = {
+    "Abundance-based Coverage Estimator (ACE) metric": "ace",
+    "Berger-Parker dominance index": "berger_parker_d",
+    "Brillouin's index": "brillouin_d",
+    "Chao1 index": "chao1",
+    "Chao1 confidence interval": "chao1_ci",
+    "Dominance measure": "dominance",
+    "Effective number of species (ENS)/"
+    "Probability of intra-or interspecific encounter (PIE) metric": "enspie",
+    "Esty's confidence interval": "esty_ci",
+    "Faith's Phylogenetic Diversity": "faith_pd",
+    "Fisher's index": "fisher_alpha",
+    "Gini index": "gini_index",
+    "Good's coverage of counts": "goods_coverage",
+    "Heip's evenness measure": "heip_e",
+    "Kempton-Taylor Q index": "kempton_taylor_q",
+    "Lladser's confidence interval": "lladser_ci",
+    "Lladser's point estimate": "lladser_pe",
+    "Margalef's richness index": "margalef",
+    "McIntosh dominance index D": "mcintosh_d",
+    "McIntosh evenness index E": "mcintosh_e",
+    "Menhinick's richness index": "menhinick",
+    "Michaelis-Menten fit to rarefaction curve of obeserved OTUs":
+        "michaelis_menten_fit",
+    "Number of distinct features": "observed_otus",
+    "Number of double occurrences": "doubles",
+    "Number of single occurrences": "singles",
+    "Number of observed features, including singles and doubles": "osd",
+    "Pielou's evenness": "pielou_e",
+    "Robbins' estimator": "robbins",
+    "Shannon's index": "shannon",
+    "Simpson's index": "simpson",
+    "Simpson's evenness measure E": "simpson_e",
+    "Strong's dominance index (Dw)": "strong"}
+
+ALPHA_CORRELATION_METHODS = {
+    "Spearman": "spearman",
+    "Pearson": "pearson"}
+
+
+BETA_DIVERSITY_METRICS = {
```

Comment:
where do these come from? can't this be an import? I'm concerned about the verifiability. Was a bit surprised to not see metrics like hellinger?

Reply:
Do we really want to integrate GUI and processing logic? We cannot verify that "braycurtis" is actually a thing that works? If not, will the strategy be to wait for a user to try a metric and hope for no failures?

Reply:
Do you know how to retrieve the metrics from the Qiime 2 plugin? It will be easy to add a test that check that all values in this dict is available in the Q2 plugin if we can retrieve that list programatically. Note that it will be a programming error and that's why I think adding a unit test is enough.

Reply:
Would be best to inquire with q2 devs

Reply:
Thanks @wasade 

I added a specific unit test that checks that all metrics/methods are exposed to the interface.

Reply:
Awesome, thanks!